src/Template.js

```diff
@@ -733,7 +733,9 @@ class Template extends TemplateContent {
 		}
 
 		// https://github.com/11ty/eleventy/issues/1206
-		data.page.rawInput = rawInput;
+		data.page.rawInput = JSON.parse(
+			JSON.stringify(rawInput, (_, v) => (typeof v === "function" ? v.toString() : v)),
+		);
```

Comment:
Or would it be better to simply not return anything...?

```suggestion
		data.page.rawInput = data.page.templateSyntax != "11ty.js" ? rawInput : undefined;
```